### PR TITLE
fix(ResponseType): Correctly add empty responses for 204/304

### PR DESCRIPTION
**Pull request by provokateurin**
Fixes https://github.com/nextcloud/openapi-extractor/issues/108

### CODE REVIEW

File: src/ResponseType.php [hunk 1]
```diff
@@ -258,14 +258,29 @@ public static function resolve(string $context, TypeNode $obj): array {
 					$contentTypes = $contentTypes !== [] ? $contentTypes : [$type != null ? "*/*" : null];
 
 					foreach ($statusCodes as $statusCode) {
-						foreach ($contentTypes as $contentType) {
+						if ($statusCode === 204 || $statusCode === 304) {
+							$customHeaders = array_filter(array_keys($headers), static fn (string $header) => str_starts_with(strtolower($header), 'x-'));
+							if (!empty($customHeaders)) {
+								Logger::error($context, 'Custom headers are not allowed: '. implode(', ', $customHeaders));
```
Comment (nickvergessen) [inline src/ResponseType.php:264]:
```suggestion
								Logger::error($context, 'Custom headers are not allowed for responses with status code 204 and 304. Found: '. implode(', ', $customHeaders));
```

**Comment by nickvergessen** [comment]
Rebased because it lost the automerge race

**Comment by provokateurin** [comment]
Seems like the notification app has a problem: `Endpoint#listNotifications: Custom headers are not allowed for responses with status code 204 or 304. Found: X-Nextcloud-User-Status`

**Comment by provokateurin** [comment]
Actually custom headers are just not allowed for 304, but 204 allows them. The error message in the server is wrong as well.

File: tests/lib/Controller/SettingsController.php [hunk 2]
```diff
@@ -439,4 +439,26 @@ public function throwingOCS(): DataResponse {
 	public function throwingOther(): DataResponse {
 		throw new NotFoundException();
 	}
+
+	/**
+	 * A route 204 response
+	 *
+	 * @return DataResponse<Http::STATUS_NO_CONTENT, array<empty>, array{}>
```
Comment (nickvergessen) [inline tests/lib/Controller/SettingsController.php:446]: could have added a header to sample/test it?

**Pull request merged**